apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+        return settingsLength - 1; // an initial setting is created during initialization, thus length will be always greater than 0
+    }
+
+    /**
+    * @dev Tell the information related to a signer
+    * @param _signer Address being queried
+    * @return lastSettingIdSigned Identification number of the last agreement setting signed by the signer
+    * @return mustSign Whether the requested signer needs to sign the current agreement setting before submitting an action
+    */
+    function _getSigner(address _signer) internal view returns (uint256 lastSettingIdSigned, bool mustSign) {
+        lastSettingIdSigned = lastSettingSignedBy[_signer];
+        mustSign = lastSettingIdSigned < _getCurrentSettingId();
+    }
+
+    /**
+    * @dev Tell the settings applicable for an action
+    * @param _action Action instance to query
+    * @return title String indicating a short description
+    * @return content Link to a human-readable text that describes the initial rules for the Agreements instance
+    * @return arbitrator Address of the IArbitrator that will be used to resolve disputes
+    */
+    function _getSettingFor(Action storage _action) internal view returns (string title, bytes content, IArbitrator arbitrator) {
+        uint256 settingId = _action.settingId;
+        require(settingId < settingsLength, ERROR_MISSING_AGREEMENT_SETTING);
+
+        Setting storage setting = settings[settingId];
+        title = setting.title;
+        content = setting.content;
+        arbitrator = setting.arbitrator;
+    }
+
+    /**
+    * @dev Tell the disputable-related information about an action
+    * @param _action Action instance being queried
+    * @return disputable Disputable app associated with the action
+    * @return requirement Collateral requirements applicable to the action
+    */
+    function _getDisputableFor(Action storage _action) internal view
+        returns (
+            IDisputable disputable,
+            CollateralRequirement storage requirement
+        )
+    {
+        disputable = _action.disputable;
+        uint256 collateralId = _action.collateralId;
+        DisputableInfo storage disputableInfo = disputableInfos[address(disputable)];
+        requirement = disputableInfo.collateralRequirements[collateralId];
+        require(collateralId < disputableInfo.collateralRequirementsLength, ERROR_MISSING_COLLATERAL_REQUIREMENT);
+    }
+
+    /**
+    * @dev Tell whether a challenge exists for an action
+    * @param _actionId Identification number of the action being queried
+    * @param _challengeId Identification number of the challenge being queried
+    * @param _challenge Challenge instance associated with the challenge identification number
+    * @return True if the requested challenge exists, false otherwise
+    */
+    function _existChallenge(uint256 _actionId, uint256 _challengeId, Challenge storage _challenge) internal view returns (bool) {
```

Comment:
Maybe better to rename as `_existChallengeForAction()` to avoid an override that doesn't quite do the same thing.

Reply:
This function was inlined